File: tms-backend/src/main/java/com/tms/dto/response/FleetInsightsResponse.java
package com.tms.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FleetInsightsResponse {
    private String engine;
    private double vehicleUtilizationPercent;
    private double maintenanceLoadPercent;
    private double driverAvailabilityPercent;
    private double expensePerTrip;
    private int fleetHealthScore;
    private String riskLevel;
    @Builder.Default
    private List<String> alerts = new ArrayList<>();
    @Builder.Default
    private List<String> recommendations = new ArrayList<>();
}

File: tms-dotnet-reports/Models/FleetInsightsResponse.cs
namespace TmsReports.Models;

public class FleetInsightsResponse
{
    public string Engine { get; set; } = "ASP.NET Core 8";
    public double VehicleUtilizationPercent { get; set; }
    public double MaintenanceLoadPercent { get; set; }
    public double DriverAvailabilityPercent { get; set; }
    public double ExpensePerTrip { get; set; }
    public int FleetHealthScore { get; set; }
    public string RiskLevel { get; set; } = "LOW";
    public List<string> Alerts { get; set; } = new();
    public List<string> Recommendations { get; set; } = new();
}
